hack/validate-owners.sh

```diff
@@ -7,15 +7,25 @@ set -o nounset
 set -o pipefail
 
 NO_OWNERS="$(mktemp)"
-trap 'rm -f $NO_OWNERS' EXIT
+WHITELIST="$(mktemp)"
+trap 'rm -f $NO_OWNERS $WHITELIST' EXIT
+
+cat > "$WHITELIST" << EOF
+.*/heketi/heketi$
+.*/openshift/kubernetes-metrics-server$
+.*/openshift/origin-web-console-server$
+.*/openshift/origin-metrics$
+.*/openshift/cluster-storage-operator$
+.*/openshift/console$
+.*/openshift/origin-web-console$
+.*/openvswitch/ovn-kubernetes$
+EOF
 
 REPOROOT="$( dirname "${BASH_SOURCE[0]}" )/.."
 
 pushd "$REPOROOT" > /dev/null
-
-find ci-operator/config -mindepth 2  -type d '!' -exec test -e "{}/OWNERS" ';' -print > "$NO_OWNERS"
-find ci-operator/jobs -mindepth 2  -type d '!' -exec test -e "{}/OWNERS" ';' -print >> "$NO_OWNERS"
-
+find ci-operator/config -mindepth 2  -type d '!' -exec test -e "{}/OWNERS" ';' -print | ( grep -v -E -f "$WHITELIST" || true ) > "$NO_OWNERS"
```

Comment:
I generally find that it's more maintainable and ultimately more readable for others if we write Bash as if it were Python ...

```sh
whitelist=(
	"heketi/heketi"
	"openshift/kubernetes-metrics-server"
	"openshift/origin-web-console-server"
	"openshift/origin-metrics"
	"openshift/cluster-storage-operator"
	"openshift/console"
	"openshift/origin-web-console"
	"openvswitch/ovn-kubernetes"
)

function whitelist_contains() {
	local org="$1"
	local repo="$2"
	for item in "${whitelist[@]}"; do
		if [[ "${item}" == "${org}/${repo}" ]]; then
			return 0
		fi
	done
	return 1
}

config="$( dirname "${BASH_SOURCE[0]}" )/config"

missing=()
present=()
for org_dir in $( find "${config}" -mindepth 1 -maxdepth 1 -type d ); do
	org="$( basename "${org_dir}" )"
	for repo_dir in $( find "${config}/${org}" -mindepth 1 -maxdepth 1 -type d ); do
		repo="$( basename "${repo_dir}" )"
		if [[ ! -s "${config}/${org}/${repo}/OWNERS" ]]; then
			if ! whitelist_contains "${org}" "${repo}"; then
				# no OWNERS and not in the whitelist is a problem
				missing+=( "${org}/${repo}" )
			fi
		elif whitelist_contains "${org}" "${repo}"; then
			# OWNERS and in the whitelist is a problem
			present+=( "${org}/${repo}" )
		fi
	done
done

failed="false"
if [[ "${#missing[@]}" -gt 0 ]]; then
	failed="true"
	cat << EOF
[ERROR] This check enforces that component configuration files are accompanied by OWNERS
[ERROR] files so that the appropriate team is able to self-service further configuration
[ERROR] change pull requests.

[ERROR] Run the following script to fetch OWNERS files from the component repositories:

[ERROR] $ ci-operator/populate-owners.sh

[ERROR] Please note that the script populates *all* ci-operator subdirectories, and it
[ERROR] takes a long time to execute (tens of minutes). If the target repository does not
[ERROR] contain an OWNERS file, it will need to be created manually.

[ERROR] The following component config directories do not have OWNERS files:
[ERROR] ${missing[*]}
EOF
fi

if [[ "${#present[@]}" -gt 0 ]]; then
	failed="true"
	cat << EOF
[ERROR] A directory that was previously whitelisted as not containing an OWNERS file is
[ERROR] now containing the file, so it no longer needs to be whitelisted. Please remove
[ERROR] the appropriate line from hack/validate-owners.sh script.

[ERROR] Components to be removed from whitelist:
[ERROR] ${present[*]}
EOF
fi

if [[ "${failed}" == "true" ]]; then
	exit 1
fi
```